src/coord/src/coord.rs

```diff
@@ -4199,13 +4208,15 @@ impl Coordinator {
         let timestamp_base = self.get_local_write_ts();
         let mut updates_by_id = HashMap::<GlobalId, Vec<Update>>::new();
         for tu in updates.into_iter() {
-            let timestamp = timestamp_base + tu.timestamp_offset;
-            for u in tu.updates {
-                updates_by_id.entry(u.id).or_default().push(Update {
-                    row: u.row,
-                    diff: u.diff,
-                    timestamp,
-                });
+            let timestamp: Option<Timestamp> = timestamp_base.results_in(&tu.timestamp_offset);
```

Comment:
I added some usages of `results_in`, which I hadn't seen before. Is this the correct way to use it?

Reply:
Ahhhh. It's surprising to see this. The `PathSummary` trait is until now largely internal to `timely`, where it is used as a lower bound on an "action that must be applied to a timestamp". It is used to summarize what must happen to timestamps as they move along paths.

I hadn't thought to use it before as "stand-in for `+` on timestamps" which I think is what you are using it for? It might be ok, but it's pretty weird. I'd be tempted to say "let's just implement `Add` for `mz_repr::Timestamp`" instead just for clarity.

Reply:
This whole PR is output from a real attempt that I thought could be applied in a smaller batch. This + example is one of many operators. We also subtract (`upper-1` in determine_timestamp), and divide and multiply (compaction time calculation). If we can implement Add, we perhaps need the other 3 also unless we want to make those operations have non-timely operations on them, which maybe defeats the purpose.